frontend/src/hooks/useNotifications.ts

```
@@ -0,0 +1,198 @@
+
+  const addAnnouncement = useCallback(
+    (title: string, message: string): string =>
+      addNotification({ title, message, category: 'announcement', type: 'info' }),
+    [addNotification]
+  );
+
+  // ── Browser push ───────────────────────────────────────────────────────────
+
+  const browserPermission = useMemo(
+    () => getBrowserPermission(),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

ESLint

[error] Definition for rule 'react-hooks/exhaustive-deps' was not found.

(react-hooks/exhaustive-deps)